firebase-database/src/main/java/com/google/firebase/database/core/utilities/Utilities.java

```diff
@@ -70,36 +44,68 @@ public static ParsedUrl parseUrl(String url) throws DatabaseException {
       if (port != -1) {
         repoInfo.secure = scheme.equals("https");
         repoInfo.host += ":" + port;
+      } else if (repoInfo.host.equals("localhost")) {
+        repoInfo.secure = scheme.equals("https");
       } else {
         repoInfo.secure = true;
       }
-      String[] parts = repoInfo.host.split("\\.", -1);
 
-      repoInfo.namespace = parts[0].toLowerCase();
+      String namespaceParam = uri.getQueryParameter("ns");
+      String[] parts = uri.getHost().split("\\.", -1);
+      if (parts.length == 3 || namespaceParam == null) {
```

Comment:
According to the internal doc, this should really test for `.firebaseio.com` instead of number of parts to align with other SDKs. Plus this probably will have some false positives for `my.dev.local` if people wants to put that in their hosts file.

Reply:
The original code is very forgiving (in my opinion much too forgiving), but making it less forgiving might break some users.

Reply:
Okay I think this is fine.